File: 10/main.py
from tqdm import tqdm
import math

sum = 0

def is_prime(num):
    ind = long(2)
    while ind < math.sqrt(num):
        if not num % ind:
            return False
        ind += 1
    return True

with tqdm(total = 2000000) as bar:
    for a in range(2000000):
        if is_prime(a):
            sum += a
        bar.update(1)

print sum

File: 10/main.rs


fn main() {
    let mut sum: u64 = 3;

    for num in 4..2000001 {
        if test_if_prime(num) {
            sum += num;
        }
    }
    println!("{}",sum)
}

fn test_if_prime(num: u64) -> bool{
    for a in 2..num {
        if num % a == 0 {
            return false;
        }
    }
    return true;
}
